src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaseEquivalences.cs

```diff
@@ -0,0 +1,167 @@
+{
+    /// <summary>
+    /// This class will perform fast lookups for case-insensitive searches in order to find which characters should
+    /// be considered equivalent. The mappings are generated based on Unicode ToLower() mappings and we generate the table
+    /// using the GenerateRegexCasingTable program located under the Tools folder for this library.
+    /// </summary>
+    internal static partial class RegexCaseEquivalences
+    {
+
+        private static char[] s_specialCasingSetBehaviors { get; } = new char[5]
+        {
+            'I', 'i', '\u0130', 'I', '\u0131'
+        };
+
+        /// <summary>
+        /// Performs a fast lookup which determines if a character is involved in case conversion, as well as
+        /// returns the characters that should be considered equivalent in case it does participate in case conversion.
+        /// This method, in contrast to <see cref="TryFindCaseEquivalencesForChar(char, out ReadOnlySpan{char})"/> does take in
+        /// culture and will also factor in the current culture in order to handle the special cases which are different between cultures.
+        /// </summary>
+        /// <param name="c">The character being analyzed</param>
+        /// <param name="culture">The <see cref="CultureInfo"/> to be used to determine the equivalences.</param>
+        /// <param name="equivalences">If <paramref name="c"/> is involved in case conversion, then equivalences will contain the
+        /// span of character which should be considered equal to <paramref name="c"/> in a case-insensitive comparison.</param>
+        /// <returns><see langword="true"/> if <paramref name="c"/> is involved in case conversion; otherwise, <see langword="false"/></returns>
+        public static bool TryFindCaseEquivalencesForCharWithIBehavior(char c, CultureInfo culture, out ReadOnlySpan<char> equivalences)
+        {
+            if ((c | 0x20) == 'i' || (c | 0x01) == '\u0131')
+            {
+                RegexCaseBehavior mappingBehavior = GetRegexBehavior(culture);
+                equivalences = c switch
+                {
+                    // Invariant mappings
+                    'i' or 'I' when mappingBehavior is RegexCaseBehavior.Invariant => s_specialCasingSetBehaviors.AsSpan(0, 2), // 'I' and 'i'
+
+                    // Non-Turkish mappings
+                    'i' or 'I' or '\u0130' when mappingBehavior is RegexCaseBehavior.NonTurkish => s_specialCasingSetBehaviors.AsSpan(0, 3), // 'I', 'i', and '\u0130'
+
+                    // Turkish mappings
+                    'I' or '\u0131' when mappingBehavior is RegexCaseBehavior.Turkish => s_specialCasingSetBehaviors.AsSpan(3, 2), // 'I' and '\u0131'
+                    'i' or '\u0130' when mappingBehavior is RegexCaseBehavior.Turkish => s_specialCasingSetBehaviors.AsSpan(1, 2), // 'i' and '\u0130'
+
+                    // Default
+                    _ => null
+                };
+                return equivalences != null;
```

Comment:
spans are value types and can't be null.  This is actually creating a default span and comparing the ref and length to see if they're equal.  The default case above should instead be `_ => default`, and this should be `return !equivalences.IsEmpty;`

Reply:
Oh, I missed that. I will fix in all places where I'm doing this.